lib/presentation/bloc/button/login_cubit.dart

```diff
@@ -0,0 +1,43 @@
+// ignore_for_file: depend_on_referenced_packages
+
+import 'package:bloc/bloc.dart';
+import 'package:injectable/injectable.dart';
+import 'package:truckbill/domain/auth/repository/auth_repository.dart';
+import 'package:truckbill/presentation/bloc/button/login_state.dart';
+
+@injectable
+class LoginCubit extends Cubit<LoginState> {
+  final AuthRepository authRepository;
+
+  LoginCubit(this.authRepository) : super(const LoginState.initial());
+
+  Future<void> loginWithGoogle() async {
+    emit(const LoginState.loading());
+    try {
+      await authRepository.signInWithGoogle();
+      emit(const LoginState.success());
+    } catch (e) {
+      emit(LoginState.failure(errorMessage: "Error: \${e.toString()}"));
```

Comment:
tutaj żeby wyświetlić snackbar robisz
```suggestion
      dispatch(LoginAction.showErrorMessage(errorMessage: "Error: \${e.toString()}"));
```